feat : Migrating registration verification fragment to MVVM

gururani-abhishek opened the pull request:
Fixes MOBILE-69

1. used RegistrationViewModel & UserAuthRepository to handle business logic and dataLayer logic for RegistrationVerificationFragment. 
2. moved userVerify payload creation to dataLayer(i.e. in UserAuthRepository)
3. covered registerationLogic(viewModel and repository) through proper unit testing -> using jUnit and Mockito
4. handled Ui state change of RegisterationVerification through RegistrationVerificaionUiState sealed class

https://github.com/openMF/mifos-mobile/assets/59947871/49bee274-b487-43e4-b7bf-518677e2ece1



Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

gururani-abhishek committed:
feat : Migrating registration verification fragment to MVVM

gururani-abhishek commented:
Hey @jawidMuhammadi, please have a look at this whenever you find time :)

jawidMuhammadi commented on app/src/main/java/org/mifos/mobile/viewModels/RegistrationVerificationViewModel.kt:
```diff
@@ -0,0 +1,46 @@
+import org.mifos.mobile.models.register.UserVerify
+import javax.inject.Inject
+
+class RegistrationVerificationViewModel @Inject constructor(private val dataManager : DataManager?) : ViewModel() {
```
No need to create this class, I believe you could use RegistrationViewModel instead and move the verification logic over there

gururani-abhishek replied:
alright, converting it to draft and then will update this after registrationFragment is done with migration.

gururani-abhishek replied:
Done

gururani-abhishek committed:
Merge remote-tracking branch 'origin/mvvm_architecture' into registration-verification-mvvm-migration

# Conflicts:
#	app/src/main/java/org/mifos/mobile/ui/fragments/RegistrationVerificationFragment.kt

gururani-abhishek committed:
fix : RegistrationVerificationFragment migration to MVVM

1. deleted registerationVerification viewModel and moved logic to registration viewModel.
2. added unit tests for registerationVerification's verifyUser function of viewModel and repository.
3. moved UI state change logic to registerationVerificationUiState sealed class.

gururani-abhishek committed:
fix : registration verification fragment mvvm migration

shifted userVerify payload creation to datalayer(i.e. in repository)

gururani-abhishek committed:
Merge remote-tracking branch 'origin/mvvm_architecture' into registration-verification-mvvm-migration

# Conflicts:
#	app/src/main/java/org/mifos/mobile/repositories/UserAuthRepository.kt
#	app/src/main/java/org/mifos/mobile/repositories/UserAuthRepositoryImp.kt

gururani-abhishek committed:
feat : registration verification mvvm migration

-> deleted extra RegistrationVerificationUiState and used RegistrationUiState instead.

gururani-abhishek commented:
Hey @jawidMuhammadi, this is done from my side, resolved the conflicts as well. Do have a look whenever you find the bandwidth. 😃

jawidMuhammadi reviewed (approved):
LGTM! :)
